prov/util/src/util_ep.c

```diff
@@ -144,6 +222,48 @@ int ofi_endpoint_close(struct util_ep *util_ep)
 		ofi_atomic_dec32(&util_ep->rx_cq->ref);
 	}
 
+	if (util_ep->rx_cntr) {
+		fid_list_remove(&util_ep->rx_cntr->ep_list,
+				&util_ep->rx_cntr->ep_list_lock,
+				&util_ep->ep_fid.fid);
+		ofi_atomic_dec32(&util_ep->rx_cntr->ref);
+	}
+
+	if (util_ep->tx_cntr) {
+		fid_list_remove(&util_ep->tx_cntr->ep_list,
+				&util_ep->tx_cntr->ep_list_lock,
+				&util_ep->ep_fid.fid);
+		ofi_atomic_dec32(&util_ep->tx_cntr->ref);
+	}
+
+	if (util_ep->rd_cntr) {
+		fid_list_remove(&util_ep->rd_cntr->ep_list,
+				&util_ep->rd_cntr->ep_list_lock,
+				&util_ep->ep_fid.fid);
+		ofi_atomic_dec32(&util_ep->rd_cntr->ref);
+	}
+
+	if (util_ep->wr_cntr) {
+		fid_list_remove(&util_ep->wr_cntr->ep_list,
+				&util_ep->wr_cntr->ep_list_lock,
+				&util_ep->ep_fid.fid);
+		ofi_atomic_dec32(&util_ep->wr_cntr->ref);
+	}
+
+	if (util_ep->rr_cntr) {
+		fid_list_remove(&util_ep->rr_cntr->ep_list,
+				&util_ep->rr_cntr->ep_list_lock,
+				&util_ep->ep_fid.fid);
+		ofi_atomic_dec32(&util_ep->rr_cntr->ref);
+	}
+
+	if (util_ep->rw_cntr) {
+		fid_list_remove(&util_ep->rw_cntr->ep_list,
+				&util_ep->rw_cntr->ep_list_lock,
+				&util_ep->ep_fid.fid);
+		ofi_atomic_dec32(&util_ep->rw_cntr->ref);
+	}
```

Comment:
But if we bound the EP to the same counter for multiple reasons (e.g. read + write), then we try to remove it once for each binding.  Or, wait, does fid_list_remove() check for this condition?  I don't remember